src/modules/clock.cpp

```diff
@@ -1,4 +1,101 @@
 #include "modules/clock.hpp"
+#include <sstream>
+#ifdef HAVE_LANGINFO_1STDAY
+#include <langinfo.h>
+#endif
+
+using zoned_time = date::zoned_time<std::chrono::system_clock::duration>;
+
+struct waybar_time {
+  std::locale locale;
+  zoned_time ztime;
+};
+
+namespace {
+
+#ifdef HAVE_LANGINFO_1STDAY
+// Computations done similarly to Linux cal utility.
+date::weekday first_day_of_week() {
+  const int i = (std::intptr_t) nl_langinfo(_NL_TIME_WEEK_1STDAY);
+  auto ymd = date::year(i / 10000)/(i / 100 % 100)/(i % 100);
+  auto wd = date::weekday(ymd);
+  uint8_t j = *nl_langinfo(_NL_TIME_FIRST_WEEKDAY);
+  return wd + date::days(j - 1);
+}
+#endif
+
+void weekdays_header(const std::locale& locale, const date::weekday& first_dow, std::ostream& os) {
+  auto wd = first_dow;
+  do {
+    if (wd != first_dow) os << ' ';
+    Glib::ustring wd_ustring(date::format(locale, "%a", wd));
+    auto wd_len = wd_ustring.length();
+    if (wd_len > 2) {
+      wd_ustring = wd_ustring.substr(0, 2);
+      wd_len = 2;
+    }
+    const std::string pad(2 - wd_len, ' ');
+    os << pad << wd_ustring;
+  } while (++wd != first_dow);
+  os << "\n";
+}
+
+struct CachedCalendar {
+  date::year_month_day ymd;
+  std::string text;
+
+  void set(const date::year_month_day& ymd_, std::string text_) {
+    ymd = ymd_;
+    text = text_;
+  }
+};
+
+CachedCalendar cached_calendar;
```

Comment:
Didn't notice that on the first pass. There could be multiple `waybar::Clock` instances with different locale/timezone values, thus it's better to cache rendered calendar per `Clock` instance as a class field.

Reply:
Done.